src/restorer/mnemonic-restorer.ts

```diff
@@ -22,36 +22,42 @@ function restorerFromEsplora<R extends MasterPublicKey>(
     gapLimit = 20,
   }) => {
     const restoreFunc = async function(
-      nextAddrFunc: (index: number) => string
-    ): Promise<number> {
+      nextAddrFunc: (index?: number) => string | undefined
+    ): Promise<number | undefined> {
       let counter = 0;
       let index = 0;
-      let maxIndex = 0;
+      let maxIndex = undefined;
 
       while (counter < gapLimit) {
         const addr = nextAddrFunc(index);
+        if (addr === undefined) break;
         const addrHasTxs = await addressHasBeenUsed(addr, esploraURL);
-
         if (addrHasTxs) {
           maxIndex = index;
           counter = 0;
-        } else counter++;
-
+        } else {
+          counter++;
+        }
         index++;
       }
-
       return maxIndex;
     };
 
-    const lastUsedExternalIndex = await restoreFunc(
-      (index: number) =>
-        identity.getAddress(false, index).address.confidentialAddress
-    );
+    const lastUsedExternalIndex = await restoreFunc((index?: number) => {
+      if (index !== undefined) {
```

Comment:
replace by `if (index)`

Reply:
`if (index)` is not correct if index is 0